### ISSUE TITLE
Reproduction Issue: Training from Scratch vs. Pretrained Model

### ISSUE THREAD

**Issue by eclipse-1234** [lzhyu/PoseShield#10]
Thank you for releasing the PoseShield code and pretrained model. I am currently trying to reproduce the reported results, and I found a noticeable difference between the released pretrained model and my model trained from scratch.

Using the provided pretrained model, I can obtain results close to those reported in the paper. However, when I train the model from scratch, the performance is noticeably worse.
--------------------------------------------------------------------------------------------------
My training configuration is as follows:

Training Configuration:
  NUM_EPOCHS: 200
  BATCH_SIZE: 512
  LR: 0.0001
  VAL_INTERVAL: 10
  EXP_NAME: "train_basic"
  GRAD_LOSS_WEIGHT: 0.0
  TD_LOSS_WEIGHT: 1.0
  DT: 0.01

Model Configuration:
  TYPE: "ResidualMLP"
  IN_DIM: 126              # 21 joints × 6 values each
  HIDDEN_DIM: 512
  NUM_LAYERS: 12           # Match the released pre-trained model
  ACTIVATION: "relu"
----------------------------------------------------------------------------------------------------
I trained 4 models from scratch and their respective evaluation results are:

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns="http://www.w3.org/TR/REC-html40"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=PowerPoint.Slide>
<meta name=Generator content="Microsoft PowerPoint 15">
<style>

</style>
</head>

<body>

  | 1 | 2 | 3 | 4
-- | -- | -- | -- | --
collision-free rate | 0.93 | 0.932 | 0.94 | 0.936
penetration depth reduction | 0.9755 | 0.9743 | 0.9769 | 0.9758
MVD | 0.0583 | 0.0585 | 0.0589 | 0.0591

</body>

</html>

----------------------------------------------------------------------------------------------------
For comparison, the results obtained with the provided pretrained model are:

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns="http://www.w3.org/TR/REC-html40"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=PowerPoint.Slide>
<meta name=Generator content="Microsoft PowerPoint 15">
<style>

</style>
</head>

<body>

  | Pretrained
-- | --
collision-free rate | 0.962
penetration depth reduction | 0.9872
MVD | 0.0602

</body>

</html>

----------------------------------------------------------------------------------------------------
Could you please help me identify what might cause this discrepancy? In particular, are there any training settings, preprocessing steps, data-generation procedures, or other details that are not included in the released configuration that I should be aware of?

One possible factor I am wondering about is the **active-learning procedure**. The paper mentions that PoseShield adopts the active-learning strategy from N-Penetrate to improve the training data, but I could not find the corresponding active-learning code or dataset in the released repository. If the pretrained model was trained using this active-learning procedure, could you please provide the active-learning code or the resulting active-learning dataset? This would be very helpful for reproducing the reported results.

Thank you very much for your help!

**Comment by lzhyu** [comment]
Hi, thanks for the detailed experiments. Regarding the active-learning pipeline, the original research code has diverged substantially through subsequent iterations of the project and is unfortunately no longer in a clean, maintainable state that we could release as a usable implementation. At the moment, we do not plan to release that exact version of the pipeline.

That said, the active-learning procedure itself is described in detail in Appendix H of the paper. It is relatively lightweight, so it should be possible to reimplement it from the description, and current coding agents may also make reconstructing the pipeline considerably easier.

**Comment by eclipse-1234** [comment]
Thanks for the clarification! I have two follow-up questions about the active-learning procedure:

1. From your experiments, does applying active learning actually improve the final performance of PoseShield compared with training without active learning? I understand that the improvement may not be very large, but I would like to know whether it provides a measurable benefit.

2. Regarding the training dataset, I noticed that it consists of two parts: **`gt_data`** and **`augment_data`**. Could you clarify whether the active-learning procedure is applied only to `gt_data`, or whether it is applied to both `gt_data` and `augment_data`?

This would help me better understand the training pipeline and reproduce the results more faithfully. Thank you again for your help!

**Comment by lzhyu** [comment]
1. I reviewed our previous results and found no clear evidence that active learning consistently improves performance. In some experiments, it improved performance by approximately 3%, whereas in others, performance decreased.

2. The procedure is applied to samples from both subsets. We randomly select samples from each subset and optimize them toward the decision boundary.

**Comment by lzhyu** [comment]
If you would like to investigate the performance further, there are several other aspects worth examining.

For example, the [current implementation](https://github.com/lzhyu/PoseShield/blob/699ea913bcf6c635e20cedfaa77c05f9a708fce9/poseshield/pose/resolve_slsqp.py#L9) uses a heuristic constraint to keep the optimized rotation representation close to a valid 6D rotation. The network input is also normalized to ensure that it represents a valid 6D rotation. These design choices could be implemented differently and may affect the results. 

A more detailed discussion of the limitations can be found in Supplementary Section D of [the paper](https://arxiv.org/pdf/2606.29686).

**Comment by eclipse-1234** [comment]
Thank you again for pointing me toward these details. They give me several useful directions for understanding the difference between the pretrained model and my from-scratch training results.
